compiler/env/RegionProfiler.hpp

```diff
@@ -48,13 +48,20 @@ namespace TR {
 class RegionProfiler
    {
 public:
-   RegionProfiler(TR::Region &region, TR::Compilation &compilation, const char *identifier) :
+   RegionProfiler(TR::Region &region, TR::Compilation &compilation, const char *format, ...) :
       _region(region),
       _initialRegionSize(_region.bytesAllocated()),
       _initialSegmentProviderSize(_region._segmentProvider.bytesAllocated()),
-      _compilation(compilation),
-      _identifier(identifier ? identifier : "[ANONYMOUS]")
+      _compilation(compilation)
       {
+      if (_compilation.getOption(TR_ProfileMemoryRegions))
+         {
+         va_list args;
+         va_start(args, format);
+         vsnprintf(_identifier, sizeof(_identifier), format, args);
+         _identifier[sizeof(_identifier) - 1] = '\0';
```

Comment:
Should there be an ASSERT or similar here if the supplied result was truncated? Not sure if we can tell, but asserting when there is a truncation rather than silently truncating would seem desirable if it isn't a huge amount of work.

Reply:
An assert has been added based on the return value of `vsnprintf`.